runtime/opt/taupage/init.d/04-configure-network-interfaces.py

```diff
@@ -0,0 +1,226 @@
+                'status': 'available',
+                'availability-zone': zone()
+            }))
+        except Exception as e:
+            logging.exception(e)
+            sys.exit(2)
+
+        if not network_interfaces:
+            logging.error(
+                'No matching "available" network interfaces volume with name %s found.', name)
+            tries -= 1
+            if tries > 0:
+                logging.error(
+                    'Sleeping for 10 seconds and hope a network interface will become "available"')
+                time.sleep(10)
+            else:
+                sys.exit(2)
+
+    if len(network_interfaces) > 1:
+        logging.warning('More than one network interface with name %s found.', name)
+        network_interfaces.sort(key=lambda v: v.id)
+    return network_interfaces[0].id
+
+
+def wait_for_network_interface_attachment(ec2, device_index):
+    tries = 3
+
+    def get_attachment():
+        return "eth{}".format(device_index) in netifaces.interfaces()
+
+    attachment = get_attachment()
+    while not attachment:
+        attachment = get_attachment()
+        if not attachment:
+            tries -= 1
+            if tries > 0:
+                logging.error(
+                    'Sleeping for 60 seconds waiting for network interface to attach')
+                time.sleep(60)
+            else:
+                sys.exit(2)
+
+
+@retry
+def attach_network_interface(ec2, network_interface_id, device_index):
+    ec2.attach_network_interface(
+        device_index=device_index,
+        network_interface_id=network_interface_id,
+        instance_id=instance_id()
+    )
+
+
+def handle_network_interfaces(region, config):
+    ec2 = ec2_client(region)
+    network_interfaces = config.get("network_interfaces", [])
+    for index, name in enumerate(network_interfaces):
+        try:
+            device_index = index + 1
+            attach_network_interface(ec2, find_network_interface(ec2, name), device_index)
+            wait_for_network_interface_attachment(ec2, device_index)
+        except Exception as e:
+            logging.exception(e)
+            sys.exit(3)
+        logging.info("Attached interface '%s'", name)
+
+
+def process_arguments():
+    parser = argparse.ArgumentParser(
+        description='Prepares interfaces according to the description in /meta/taupage.yaml'
+    )
+
+    parser.add_argument(
+        '-f',
+        '--file',
+        dest='filename',
+        default='/meta/taupage.yaml',
+        help='config file in YAML'
+    )
+
+    parser.add_argument(
+        '-d',
+        '--debug',
+        action='store_true',
+        help='log additional info, for debugging purposes'
+    )
+
+    parser.add_argument(
+        '-r',
+        '--region',
+        dest='region',
+        help='uses a specific AWS region instead of querying the instance metadata'
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = process_arguments()
```

Comment:
the taupage init script is called without any arguments:
https://github.com/zalando-stups/taupage/blob/master/runtime/opt/taupage/init.sh#L24

So have this argument parsing here might be overkill.

If you want to have this script available for calling it manually (with different arguments/options), maybe place it in `runtime/opt/taupage/bin/` and make the one in `/opt/taupage/init.d/` a bash script just calling the one in bin.

Reply:
Again comes what can be seen in the volume script: https://github.com/zalando-stups/taupage/blob/master/runtime/opt/taupage/init.d/10-prepare-disks.py#L418

Reply:
Ok, ignore my comment then. :)